database/migrate.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+function missing(dep) {
+	try {
+		require.resolve(dep);
+		return false;
+	} catch (err) {
+		if (err.code !== 'MODULE_NOT_FOUND') throw err;
+		return true;
+	}
+}
+
+function shell(cmd) {
+	child_process.execSync(cmd, {stdio: 'inherit', cwd: __dirname});
+}
+
+if (missing('sqlite')) {
```

Comment:
I would have just inlined `missing` and `shell` here because they're not used more than once, but I guess it reads better? It also makes me wonder if we should have a common tool-utils library somewhere given I obviously have copied and pasted these functions all over the place at this point?

Reply:
> It also makes me wonder if we should have a common tool-utils library somewhere

Possibly. That's out of scope for this PR though.